managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpReadEntry.java

```diff
@@ -60,7 +61,18 @@ public static OpReadEntry create(ManagedCursorImpl cursor, PositionImpl readPosi
             maxPosition = PositionImpl.LATEST;
         }
         op.maxPosition = maxPosition;
-        op.skipCondition = skipCondition;
+
+        Predicate<PositionImpl> skipCondition0 = cursor instanceof ReadOnlyCursor ? null : cursor::isMessageDeleted;
+        if (skipCondition == null) {
+            op.skipCondition = skipCondition0;
+        } else {
+            if (skipCondition0 == null) {
+                op.skipCondition = skipCondition;
+            } else {
+                op.skipCondition = skipCondition.or(skipCondition0);
+            }
+        }
```

Comment:
I think it's better to pass a parameter to specify whether to skip deleted messages or pass in skipCondition instead of changing the behavior of read directly.

Reply:
> I think it's better to pass a parameter to specify whether to skip deleted messages or pass in skipCondition instead of changing the behavior of read directly.

For `OpReadEntry` it will filter-out deleted messages in `readEntriesComplete`, so I think add `skipCondition` directly is makes sense, It should be consistent with the original semantics